discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/Discovery.java

```diff
@@ -76,379 +45,803 @@
-        supportedMediaTypes
-                .put(HttpMediaType.APPLICATION_MS_WORD_DOCX, MediaType.parse(HttpMediaType.APPLICATION_MS_WORD_DOCX));
-        supportedMediaTypes.put(HttpMediaType.APPLICATION_PDF, MediaType.parse(HttpMediaType.APPLICATION_PDF));
-        supportedMediaTypes.put(HttpMediaType.TEXT_HTML, MediaType.parse(HttpMediaType.TEXT_HTML));
-        supportedMediaTypes
-                .put(HttpMediaType.APPLICATION_XHTML_XML, MediaType.parse(HttpMediaType.APPLICATION_XHTML_XML));
-
-        List<String> operatorsToEscape = new ArrayList<String>();
-        for (Operator operator : Operator.values()) {
-            if (operator.shouldEscape()) {
-                operatorsToEscape.add(Pattern.quote(operator.getSymbol()));
-            }
-        }
-
-        escapeControlCharactersPattern = Pattern.compile("(" + RequestUtils.join(operatorsToEscape, "|") + ")");
+public class Discovery extends WatsonService {
+
+  private static final String SERVICE_NAME = "discovery";
+  private static final String URL = "https://gateway.watsonplatform.net/discovery/api";
+
+  private String versionDate;
+
+  /** The Constant VERSION_DATE_2016_12_01. */
+  public static final String VERSION_DATE_2016_12_01 = "2016-12-01";
+
+  /**
+   * Instantiates a new `Discovery`.
+   *
+   * @param versionDate The version date (yyyy-MM-dd) of the REST API to use. Specifying this value will keep your API
+   *                    calls from failing when the service introduces breaking changes.
+   */
+  public Discovery(String versionDate) {
+    super(SERVICE_NAME);
+    if ((getEndPoint() == null) || getEndPoint().isEmpty()) {
+      setEndPoint(URL);
     }
 
-    /**
-     * Instantiates a new Discovery service.
-     *
-     * @param versionDate the version date
-     */
-    public Discovery(String versionDate) {
-        super(SERVICE_NAME);
-        this.versionDate = versionDate;
+    Validator.isTrue((versionDate != null) && !versionDate.isEmpty(),
+        "'version cannot be null. Use " + VERSION_DATE_2016_12_01);
+
+    this.versionDate = versionDate;
+  }
+
+  /**
+   * Instantiates a new `Discovery` with username and password.
+   *
+   * @param versionDate The version date (yyyy-MM-dd) of the REST API to use. Specifying this value will keep your API
+   *                    calls from failing when the service introduces breaking changes.
+   * @param username    the username
+   * @param password    the password
+   */
+  public Discovery(String versionDate, String username, String password) {
+    this(versionDate);
+    setUsernameAndPassword(username, password);
+  }
+
+  /**
+   * Create a collection.
+   *
+   * @param environmentId the ID of your environment
+   * @param body          Input a JSON object that allows you to add a collection.
+   * @return the {@link Collection} with the response
+   */
+  public ServiceCall<Collection> createCollection(String environmentId, CreateCollectionRequest body) {
```

Comment:
The alphabetical ordering is imposed by the base [swagger-codegen](https://github.com/swagger-api/swagger-codegen) generator that is not currently configurable and would not be a small effort to fix.  We've chosen to accept this behavior for now because -- the thinking is -- most developers won't actually "read" the SDK files but rather will use code completion and such, which is not impacted by the order of operations in the SDK file.